lib/matplotlib/legend.py

```diff
@@ -480,6 +483,22 @@ def val_or_rc(val, rc_name):
         self._mode = mode
         self.set_bbox_to_anchor(bbox_to_anchor, bbox_transform)
 
+        # Figure out if self.shadow is valid
+        # If shadow was None, rcParams loads False
+        # So it shouldn't be None here
+
+        if isinstance(self.shadow, dict):
```

Comment:
I think it makes sense to initialize `self._shadow_props` with `{'ox': -2, 'oy': -2}` and then update below. In that way one can change the shadow offset in `shadow`.